File: Power Apps Project 1/Control Tower/Models/Flight.cs
namespace Control_Tower.Models
{
    public class Flight
    {
        public int ID { get; set; }
        public DateTime DepartureDateTime { get; set; }
        public DateTime ArrivalDateTime { get; set; }
        public string DepartureAirport { get; set; } = String.Empty;
        public string ArrivalAirport { get; set; } = String.Empty;
        public int PassengerLimit { get; set; }
    }
}

File: control-tower-frontend/src/view-models/flight.ts
// Flight model
export interface Flight
{
   id : number,
   departureDateTime : string,
   arrivalDateTime : string,
   departureAirport : string,
   arrivalAirport : string,
   passengerLimit : number
}

// Useful type for flight data queries
export type FlightDataParams =
{
   flightNumber : number | null,
   departureType: number,
   departureDateTime1 : Date | null,
   departureDateTime2 : Date | null,
   arrivalType: number,
   arrivalDateTime1 : Date | null,
   arrivalDateTime2 : Date | null,
   departureAirport : string,
   arrivalAirport : string,
   passengerLimit : number | null
}